addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1403,6 +1404,97 @@ export class Wysiwyg extends Component {
             });
         }
     }
+    /**
+     * Open the content generation dialog to generate content.
+     *
+     * @param {'prompt'|'alternatives'} [mode='prompt']
+     */
+    openGenerateContentDialog(mode = 'prompt') {
+        const onContentInserted = insertedNodes => {
+            this.notification.add(_t('Your content was successfully generated.'), {
+                title: _t('Content generated'),
+                type: 'success',
+            });
+            const start = insertedNodes?.length && closestElement(insertedNodes[0]);
+            const end = insertedNodes?.length && closestElement(insertedNodes[insertedNodes.length - 1]);
+            if (start && end) {
+                const div = document.createElement('div');
+                div.style.left = start.offsetLeft - 10 + 'px';
+                div.style.top = start.offsetTop - 10 + 'px';
+                div.style.width = Math.max(start.clientWidth, end.clientWidth) + 20 + 'px';
+                div.style.height = end.offsetTop + end.clientHeight - start.offsetTop + 20 + 'px';
+                div.classList.add('o_generated_content');
+                this.odooEditor.editable.parentElement.prepend(div);
+                setTimeout(() => div.remove(), 2000);
```

Comment:
What's the matter? It's a temporary element that's there to show where the content was inserted for 2s, then it goes away.

Reply:
I had no way to know that because there is no comment to explain what this code is doing. Having this helper ui was a very deliberate decision, is not something you might expect from this function naively, and takes 20 loc just for itself. I strongly think the feature deserves a comment.